R/utility.R

```diff
@@ -505,30 +473,73 @@ waitForTasksToComplete <- function(jobId, timeout) {
       currentTasks <-
         rAzureBatch::listTask(jobId, skipToken = URLdecode(skipTokenValue))
 
-      taskStates <-
-        lapply(currentTasks$value, function(x)
-          x$state != "completed")
+      totalTasks <- totalTasks + length(currentTasks$value)
+    }
 
-      for (i in 1:length(taskStates)) {
-        if (taskStates[[i]] == FALSE) {
-          count <- count + 1
+    pb <- txtProgressBar(min = 0, max = totalTasks, style = 3)
+    timeToTimeout <- Sys.time() + timeout
+
+    repeat {
+      taskCounts <- rAzureBatch::getJobTaskCounts(jobId)
+
+      setTxtProgressBar(pb, taskCounts$completed)
+
+
+      if (taskCounts$failed > 0 &&
+          errorHandling == "stop" &&
+          taskCounts$validationStatus == "Validated") {
+        cat("\n")
+
+        filter <- "state eq 'completed' and executionInfo/exitCode ne 0"
```

Comment:
This may not be a sufficient filter. Double check with Jun around what is considered a failure. For example, when a resource file is not found it will also fail the task, but the exit code will not be set (instead some other value is there)

Reply:
This filter does not work, for scheduling errors. Will have to do a client side filter for now...

filter <- "state eq 'completed' and executionInfo/result eq 'Failure'" is not queryable yet until we deprecate one of our service components.